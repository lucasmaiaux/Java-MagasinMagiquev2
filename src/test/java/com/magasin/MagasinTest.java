package com.magasin;

import org.approvaltests.Approvals;
import org.approvaltests.combinations.CombinationApprovals;
import org.approvaltests.combinations.CombinationsHelper;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.CsvSource;

import static org.junit.jupiter.api.Assertions.assertEquals;

class MagasinTest {

    @Test
    public void updateQuality() {
        String[] names = new String[]{"Comt√©", "Pass VIP Concert", "Kryptonite", "Truc"};
        Integer[] sellIns = new Integer[]{-1,0,1,5,10,15};
        Integer[] qualities = new Integer[]{0, 25, 55};

        CombinationApprovals.verifyAllCombinations(
                this::doUpdateQuality,
                names,
                sellIns,
                qualities
        );
    }

    private String doUpdateQuality(String name, int sell_in, int quality) {
        Item[] items = new Item[] { new Item(name, sell_in, quality)};
        Magasin app = new Magasin(items);
        app.updateQuality();
        return app.items[0].toString();
    }

    @ParameterizedTest
    @CsvSource({
            "4, 10, 3, 8",
            "10, 10, 9, 8",
            "0, 0, -1, 0"
    })
    void testMagic(int sell_in, int quality, int expected_sellin, int expected_quality) {
        Item[] items = new Item[] { new Item("Pouvoirs magiques", sell_in, quality)};
        Magasin app = new Magasin(items);
        app.updateQuality();
        assertEquals(expected_sellin, app.items[0].sellIn);
        assertEquals(expected_quality, app.items[0].quality);
    }

}
