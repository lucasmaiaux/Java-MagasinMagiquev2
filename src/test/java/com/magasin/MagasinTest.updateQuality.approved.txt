[Comté, -1, 0] => Comté, -2, 2 
[Comté, -1, 25] => Comté, -2, 27 
[Comté, -1, 55] => Comté, -2, 55 
[Comté, 0, 0] => Comté, -1, 2 
[Comté, 0, 25] => Comté, -1, 27 
[Comté, 0, 55] => Comté, -1, 55 
[Comté, 1, 0] => Comté, 0, 1 
[Comté, 1, 25] => Comté, 0, 26 
[Comté, 1, 55] => Comté, 0, 55 
[Comté, 5, 0] => Comté, 4, 1 
[Comté, 5, 25] => Comté, 4, 26 
[Comté, 5, 55] => Comté, 4, 55 
[Comté, 10, 0] => Comté, 9, 1 
[Comté, 10, 25] => Comté, 9, 26 
[Comté, 10, 55] => Comté, 9, 55 
[Comté, 15, 0] => Comté, 14, 1 
[Comté, 15, 25] => Comté, 14, 26 
[Comté, 15, 55] => Comté, 14, 55 
[Pass VIP Concert, -1, 0] => Pass VIP Concert, -2, 0 
[Pass VIP Concert, -1, 25] => Pass VIP Concert, -2, 0 
[Pass VIP Concert, -1, 55] => Pass VIP Concert, -2, 0 
[Pass VIP Concert, 0, 0] => Pass VIP Concert, -1, 0 
[Pass VIP Concert, 0, 25] => Pass VIP Concert, -1, 0 
[Pass VIP Concert, 0, 55] => Pass VIP Concert, -1, 0 
[Pass VIP Concert, 1, 0] => Pass VIP Concert, 0, 3 
[Pass VIP Concert, 1, 25] => Pass VIP Concert, 0, 28 
[Pass VIP Concert, 1, 55] => Pass VIP Concert, 0, 55 
[Pass VIP Concert, 5, 0] => Pass VIP Concert, 4, 3 
[Pass VIP Concert, 5, 25] => Pass VIP Concert, 4, 28 
[Pass VIP Concert, 5, 55] => Pass VIP Concert, 4, 55 
[Pass VIP Concert, 10, 0] => Pass VIP Concert, 9, 2 
[Pass VIP Concert, 10, 25] => Pass VIP Concert, 9, 27 
[Pass VIP Concert, 10, 55] => Pass VIP Concert, 9, 55 
[Pass VIP Concert, 15, 0] => Pass VIP Concert, 14, 1 
[Pass VIP Concert, 15, 25] => Pass VIP Concert, 14, 26 
[Pass VIP Concert, 15, 55] => Pass VIP Concert, 14, 55 
[Kryptonite, -1, 0] => Kryptonite, -1, 0 
[Kryptonite, -1, 25] => Kryptonite, -1, 25 
[Kryptonite, -1, 55] => Kryptonite, -1, 55 
[Kryptonite, 0, 0] => Kryptonite, 0, 0 
[Kryptonite, 0, 25] => Kryptonite, 0, 25 
[Kryptonite, 0, 55] => Kryptonite, 0, 55 
[Kryptonite, 1, 0] => Kryptonite, 1, 0 
[Kryptonite, 1, 25] => Kryptonite, 1, 25 
[Kryptonite, 1, 55] => Kryptonite, 1, 55 
[Kryptonite, 5, 0] => Kryptonite, 5, 0 
[Kryptonite, 5, 25] => Kryptonite, 5, 25 
[Kryptonite, 5, 55] => Kryptonite, 5, 55 
[Kryptonite, 10, 0] => Kryptonite, 10, 0 
[Kryptonite, 10, 25] => Kryptonite, 10, 25 
[Kryptonite, 10, 55] => Kryptonite, 10, 55 
[Kryptonite, 15, 0] => Kryptonite, 15, 0 
[Kryptonite, 15, 25] => Kryptonite, 15, 25 
[Kryptonite, 15, 55] => Kryptonite, 15, 55 
[Truc, -1, 0] => Truc, -2, 0 
[Truc, -1, 25] => Truc, -2, 23 
[Truc, -1, 55] => Truc, -2, 53 
[Truc, 0, 0] => Truc, -1, 0 
[Truc, 0, 25] => Truc, -1, 23 
[Truc, 0, 55] => Truc, -1, 53 
[Truc, 1, 0] => Truc, 0, 0 
[Truc, 1, 25] => Truc, 0, 24 
[Truc, 1, 55] => Truc, 0, 54 
[Truc, 5, 0] => Truc, 4, 0 
[Truc, 5, 25] => Truc, 4, 24 
[Truc, 5, 55] => Truc, 4, 54 
[Truc, 10, 0] => Truc, 9, 0 
[Truc, 10, 25] => Truc, 9, 24 
[Truc, 10, 55] => Truc, 9, 54 
[Truc, 15, 0] => Truc, 14, 0 
[Truc, 15, 25] => Truc, 14, 24 
[Truc, 15, 55] => Truc, 14, 54 
